### ISSUE TITLE
Incorrect repo representation of jump tables

### ISSUE THREAD

**Issue by paulhuggett** [SNSystems/llvm-prepo#23]
A simple switch statement such as:

```
int foo (int mode) {
    switch (mode) {
    case 1: return 5;
    case 2: return 6;
    case 3: return 7;
    case 4: return 8;
    default: return 9;
    }
}
```

Compiled to the repo with a command something like:

    clang -O0 -c -target x86_64-pc-linux-gnu-repo -o switch.o switch.c

Produces ifixup records in foo’s fragment which to do not have the correct addend value:

```
$ dumpfragment switch.o foo 
---

- file      : 
      path : ./clang.db
      size : 4194304
  fragments : 
      - digest   : c2db93450e69f48bc49f046cccaf6548
        fragment : 
            - type     : text
              contents : 
                  align   : 16
                  data    : <snip instructions>
                  ifixups : 
                      - { section: read_only, type: 11, offset: 39, addend: 0 }
                  xfixups : [ ]
            - type     : read_only
              contents : 
                  align   : 8
                  data    : <snip 32 zero bytes>
                  ifixups : 
                      - { section: text, type: 1, offset: 0, addend: 0 }
                      - { section: text, type: 1, offset: 8, addend: 0 }
                      - { section: text, type: 1, offset: 16, addend: 0 }
                      - { section: text, type: 1, offset: 24, addend: 0 }
                  xfixups : [ ]
...
```

These lines are: the interesting ones: "- { section: text, type: 1, offset: 0, **addend: 0**}".

This contrasts with the same file compiled directly to ELF:
```
$ llvm-readobj --relocations switch.elf

File: switch.elf
Format: ELF64-x86-64
Arch: x86_64
AddressSize: 64bit
LoadName: 
Relocations [
  Section (3) .rela.text {
    0x27 R_X86_64_32S .rodata 0x0
  }
  Section (5) .rela.rodata {
    0x0 R_X86_64_64 .text 0x2D
    0x8 R_X86_64_64 .text 0x39
    0x10 R_X86_64_64 .text 0x45
    0x18 R_X86_64_64 .text 0x51
  }
  Section (9) .rela.eh_frame {
    0x20 R_X86_64_PC32 .text 0x0
  }
]
```

**Comment by paulhuggett** [comment]
With commit 953151252165595f2eb84abf3a4e35ee218074ff the output has changed so that the repository contents are now correct. Here's the pstore-dump output for the same code:
```
bash-3.2$ dumpfragment switch.o foo
---

- file      : 
      path : ./clang.db
      size : 0x400000
  fragments : 
      - digest   : c2db93450e69f48bc49f046cccaf6548
        fragment : 
            - type     : text
              contents : 
                  align   : 0x10
                  data    : 
                      - 0x00:   pushq   %rbp
                      - 0x01:   movq    %rsp, %rbp
                      - 0x04:   movl    %edi, -0x8(%rbp)
                      - 0x07:   movl    -0x8(%rbp), %edi
                      - 0x0a:   addl    $-0x1, %edi
                      - 0x0d:   movl    %edi, %eax
                      - 0x0f:   subl    $0x3, %edi
                      - 0x12:   movq    %rax, -0x10(%rbp)
                      - 0x16:   movl    %edi, -0x14(%rbp)
                      - 0x19:   ja      0x3e
                      - 0x1f:   movq    -0x10(%rbp), %rax
                      - 0x23:   movq    (,%rax,8), %rcx
                      - 0x2b:   jmpq    *%rcx
                      - 0x2d:   movl    $0x5, -0x4(%rbp)
                      - 0x34:   jmp     0x2b
                      - 0x39:   movl    $0x6, -0x4(%rbp)
                      - 0x40:   jmp     0x1f
                      - 0x45:   movl    $0x7, -0x4(%rbp)
                      - 0x4c:   jmp     0x13
                      - 0x51:   movl    $0x8, -0x4(%rbp)
                      - 0x58:   jmp     0x7
                      - 0x5d:   movl    $0x9, -0x4(%rbp)
                      - 0x64:   movl    -0x4(%rbp), %eax
                      - 0x67:   popq    %rbp
                      - 0x68:   retq
                  ifixups : 
                      - { section: read_only, type: 0xb, offset: 0x27, addend: 0x0 }
                  xfixups : [ ]
            - type     : read_only
              contents : 
                  align   : 0x8
                  data    : !!binary16 |
                      0000 0000 0000 0000 0000 0000 0000 0000
                      0000 0000 0000 0000 0000 0000 0000 0000>
                  ifixups : 
                      - { section: text, type: 0x1, offset: 0x0, addend: 0x2d }
                      - { section: text, type: 0x1, offset: 0x8, addend: 0x39 }
                      - { section: text, type: 0x1, offset: 0x10, addend: 0x45 }
                      - { section: text, type: 0x1, offset: 0x18, addend: 0x51 }
                  xfixups : [ ]
...
bash-3.2$ 
```
Look for the addend values on the `read_only` section's ifixups. These now match the addends in the compiler's `.rela.rodata` section.

**Comment by paulhuggett** [comment]
Finally fixed by commit b737c6cf2b91b478a3a02f16785ad86cfa216eb5.
